### PR TITLE
[elc] add support for sorting results by SortOptions to NativeQuery

### LINKED ISSUE #2263: Support for sorting results via SortOptions of the Elasticsearch API client.
At the moment there is no way to sort the results using `SortOptions` in the `NativeQuery`, but this is necessary to perform more complex sorting.

https://www.elastic.co/guide/en/elasticsearch/reference/8.3/sort-search-results.html

### PR DESCRIPTION
**Pull request by xhaggi**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] **There is a ticket in the bug tracker for the project in our [issue tracker](https://github.
  com/spring-projects/spring-data-elasticsearch/issues)**. Add the issue number to the _Closes #issue-number_ line below
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

Closes #2263

### CODE REVIEW

File: src/main/java/org/springframework/data/elasticsearch/client/elc/NativeQueryBuilder.java [hunk 1]
```diff
@@ -129,6 +139,30 @@ public NativeQueryBuilder withScriptedField(ScriptedField scriptedField) {
...
+		sortOptions.add(value);
+		if (values.length > 0) {
+			sortOptions.addAll(Arrays.asList(values));
+		}
+
+		return this;
+	}
+
+	public NativeQueryBuilder withSort(Function<SortOptions.Builder, ObjectBuilder<SortOptions>> fn) {
+
```
Comment (sothawo) [inline src/main/java/org/springframework/data/elasticsearch/client/elc/NativeQueryBuilder.java:161]: assert that `fn` is not null

**Review by sothawo — CHANGES_REQUESTED** [review]
some minor remarks

File: src/main/java/org/springframework/data/elasticsearch/client/elc/NativeQueryBuilder.java [hunk 2]
```diff
@@ -129,6 +139,30 @@ public NativeQueryBuilder withScriptedField(ScriptedField scriptedField) {
 		return this;
 	}
 
+	public NativeQueryBuilder withSort(List<SortOptions> values) {
```
[the remark this answers is not in the document]
Comment (xhaggi) [reply src/main/java/org/springframework/data/elasticsearch/client/elc/NativeQueryBuilder.java:142]: I have no strong opinion on this, but the `BaseQueryBuilder` also had a method `withSort` and the `SearchRequest.Builder` has a method called `sort`.

File: src/main/java/org/springframework/data/elasticsearch/client/elc/NativeQueryBuilder.java (hunk 1 above)
Comment (xhaggi) [reply src/main/java/org/springframework/data/elasticsearch/client/elc/NativeQueryBuilder.java:161]: Done.

**Review by sothawo — APPROVED** [review]

**Pull request merged**